src/odc/sql/TODATableIterator.cc

```diff
@@ -23,6 +24,33 @@ namespace sql {
 
 //----------------------------------------------------------------------------------------------------------------------
 
+namespace {
+
+/// Return the index of column `columnName` in metadata. If not found or ambiguous,
+/// throw an exception.
+size_t columnIndex(const std::string& columnName, const core::MetaData &md)
+{
+    size_t idx;
+    try {
+        idx = md.columnIndex(columnName);
+    } catch (const eckit::UserError &e) {
+        // Make some error messages more precise
+        if (eckit::StringTools::startsWith(e.what(), "Ambiguous column name:"))
```

Comment:
This is something of a code smell.
By preference, I would write the inner exceptions such that they don't need to be wrapped.
But if the message cannot be written sufficiently generically, such that you really need to put the "SQL request" text in here, please define derived exception types, and then catch them specifically. Looking at the string contents is something of a smell, and is very much more brittle.